packages/build/src/commands/plugin.js

```diff
@@ -73,8 +71,7 @@ const updateNetlifyConfig = async function ({
   }
 
   const priorityConfigA = applyMutations(priorityConfig, configMutations)
-  const originalConfig = await resolveUpdatedConfig(configOpts)
-  const netlifyConfigA = deepmerge(originalConfig, priorityConfigA)
+  const netlifyConfigA = await resolveUpdatedConfig(configOpts, priorityConfigA)
```

Comment:
Let `@netlify/config` perform the merge instead using its `priorityConfig` option.